lbz/configuration/managers.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Dict, List
+
+from lbz.configuration.configs import BaseConfig
+from lbz.exceptions import ConfigurationMissingKey
+
+
+class PreLoadConfigManager:
+    def __init__(self, configs: List[BaseConfig]) -> None:
+        self.configs: Dict[str, BaseConfig] = {cfg.key: cfg for cfg in configs}
+        self.loaded_configs: Dict[str, Any] = {}
+        for cfg in configs:
+            value = cfg.get_value()
+            if value is cfg.default is None:
+                raise ConfigurationMissingKey(cfg.key)
+            self.loaded_configs[cfg.key] = value
+
+    def __getattr__(self, item: str) -> Any:
+        return self.loaded_configs[item]
```

Comment:
Now, I understand why you mentioned the need of having `.pyi` modules 😉 

I need to say that this manager won't be easy to use/maintain, so, I think we should consider a replacement if this is intended to become a core logic in every single other repository 😬